x/ibc/02-client/client/cli/tx.go

```diff
@@ -74,35 +74,47 @@ $ %s tx ibc client create [client-id] [path/to/consensus_state.json] --from node
 // https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#update
 func GetCmdUpdateClient(cdc *codec.Codec) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "update [client-id] [path/to/header.json]",
+		Use:   "update [client-id] [path/to/old-header.json] [path/to/new-header.json]",
 		Short: "update existing client with a header",
 		Long: strings.TrimSpace(fmt.Sprintf(`update existing client with a header:
 
 Example:
 $ %s tx ibc client update [client-id] [path/to/header.json] --from node0 --home ../node0/<app>cli --chain-id $CID
 		`, version.ClientName),
 		),
-		Args: cobra.ExactArgs(2),
+		Args: cobra.ExactArgs(3),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			inBuf := bufio.NewReader(cmd.InOrStdin())
 			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(authclient.GetTxEncoder(cdc))
 			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
 
 			clientID := args[0]
 
-			var header exported.Header
-			if err := cdc.UnmarshalJSON([]byte(args[1]), &header); err != nil {
+			var oldHeader, newHeader exported.Header
+
+			if err := cdc.UnmarshalJSON([]byte(args[1]), &oldHeader); err != nil {
 				// check for file path if JSON input is not provided
 				contents, err := ioutil.ReadFile(args[1])
 				if err != nil {
 					return errors.New("neither JSON input nor path to .json file were provided")
 				}
-				if err := cdc.UnmarshalJSON(contents, &header); err != nil {
+				if err := cdc.UnmarshalJSON(contents, &oldHeader); err != nil {
+					return errors.Wrap(err, "error unmarshalling header file")
+				}
+			}
+
+			if err := cdc.UnmarshalJSON([]byte(args[2]), &newHeader); err != nil {
```

Comment:
I think this would fall back to reading a file if e.g. invalid JSON was given as well. It might be better to have some heuristic for detecting JSON (e.g. look for `{` as first character), or having an option for it.

Reply:
Agreed, this is something we do throughout the CLI and at this point we could likely write some concise logic and share it through the codebase.